site/src/api/api.ts

```diff
@@ -1017,12 +1019,22 @@ const getMissingParameters = (
   templateParameters: TypesGen.TemplateVersionParameter[],
 ) => {
   const missingParameters: TypesGen.TemplateVersionParameter[] = []
-  const requiredParameters = templateParameters.filter(
-    // It is required
-    // and it can be changed
-    // and it is not from a legacy variable
-    (p) => p.required && p.mutable && p.legacy_variable_name === undefined,
-  )
+  const requiredParameters: TypesGen.TemplateVersionParameter[] = []
+
+  templateParameters.forEach((p) => {
+    // Legacy parameters should be required. So we can migrate them.
+    const isLegacy = p.legacy_variable_name === undefined
+    // It is mutable and required. Mutable values can be changed after so we
+    // don't need to ask them if they are not required.
+    const isMutableAndRequired = p.mutable && p.required
+    // Is immutable, so we can check if it is its first time on the build
+    const isImmutable = !p.mutable
+
+    if (isLegacy || isMutableAndRequired || isImmutable) {
+      requiredParameters.push(p)
+      return
+    }
+  })
```

Comment:
@mtojek @bpmct is this logic correct? I must admit that asking or not for missing parameters is pretty challenging to understand for me 😅

Reply:
Let me try it locally...